src/test/java/org/prebid/server/analytics/reporter/greenbids/GreenbidsAnalyticsReporterTest.java

```diff
@@ -0,0 +1,468 @@
+import org.prebid.server.version.PrebidVersionProvider;
+import org.prebid.server.vertx.httpclient.HttpClient;
+import org.prebid.server.vertx.httpclient.model.HttpClientResponse;
+
+import java.math.BigDecimal;
+import java.time.Clock;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import static java.util.Collections.singletonMap;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class GreenbidsAnalyticsReporterTest extends VertxTest {
+
+    @Rule
+    public final MockitoRule mockitoRule = MockitoJUnit.rule();
+
+    @Mock
+    private HttpClient httpClient;
+
+    @Mock
+    private Clock clock;
+
+    @Mock
+    private PrebidVersionProvider prebidVersionProvider;
+
+    private GreenbidsAnalyticsReporter target;
+
+    private AuctionEvent event;
+
+    private GreenbidsAnalyticsProperties greenbidsAnalyticsProperties;
+
+    @Before
+    public void setUp() {
+        final ObjectMapper mapper = ObjectMapperProvider.mapper();
+        jacksonMapper = new JacksonMapper(mapper);
+
+        greenbidsAnalyticsProperties = GreenbidsAnalyticsProperties.builder()
+                .exploratorySamplingSplit(0.9)
+                .analyticsServerVersion("2.2.0")
+                .analyticsServer("http://localhost:8080")
+                .configurationRefreshDelayMs(10000L)
+                .timeoutMs(100000L)
+                .build();
+
+        target = new GreenbidsAnalyticsReporter(
+                greenbidsAnalyticsProperties,
+                jacksonMapper,
+                httpClient,
+                clock,
+                prebidVersionProvider);
+    }
+
+    @Test
+    public void shouldReceiveValidResponseOnAuctionContextForBanner() {
+        // given
+        final Banner banner = setUpBanner();
+
+        final ObjectNode prebidJsonNodes = mapper.valueToTree(
+                singletonMap("gpid", TextNode.valueOf("gpidvalue")));
+
+        final Imp imp = Imp.builder()
+                .id("adunitcodevalue")
+                .ext(prebidJsonNodes)
+                .banner(banner)
+                .build();
+
+        final AuctionContext auctionContext = setupAuctionContext(imp);
+        event = AuctionEvent.builder()
+                .auctionContext(auctionContext)
+                .bidResponse(auctionContext.getBidResponse())
+                .build();
+
+        final HttpClientResponse mockResponse = mock(HttpClientResponse.class);
+        when(mockResponse.getStatusCode()).thenReturn(202);
+        when(httpClient.post(anyString(), any(MultiMap.class), anyString(), anyLong()))
```

Comment:
As I can judge sending the correct message is the most important part here, but you never actually check the correctness of the data, please cover it as well

Reply:
I've added checks of the json content 

Here I've used `ArgumentCaptor<String>`.  Both jsons `capturedJson` and `expectedJson` are compared to each other ( random UUIDs `greenbidsId`, `billingId` are set manually).